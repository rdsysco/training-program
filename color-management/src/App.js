import React, {Component} from 'react';
import './App.css';
import {Card, CardAdd} from './components'

class App extends Component {

    constructor() {

        super();

        this.state = {
            items: [
                {
                    title: 'Inna',
                    color: 'red',
                    rate: 4
                },
                {
                    title: 'Selena Gomez',
                    color: 'yellow',
                    rate: 3
                },
                {
                    title: 'Jennifer Lopez',
                    color: 'blue',
                    rate: 1
                }
            ]
        }
    }

    add(value) {

        const items = this.state.items || [];

        items.push(value);

        this.setState({
            items
        })
    }

    change(index, value) {

        const items = this.state.items || [];

        items.splice(index, 1, value);

        this.setState({
            items
        })
    }

    remove(index) {

        if (!window.confirm('Are you sure to delete?')) return;

        const items = this.state.items || [];

        items.splice(index, 1);

        this.setState({
            items
        })
    }

    render() {
        return (
            <div>
                <div style={{textAlign: 'center', padding: '15px'}}>
                    <CardAdd onAdd={(value) => this.add(value)}/>
                </div>
                <div>
                    {
                        this.state.items.map((item, index) => {

                            return (
                                <Card key={index}
                                      value={item}
                                      onChange={(value) => this.change(index, value)}
                                      onRemove={() => this.remove(index)}/>
                            );
                        })
                    }
                </div>
            </div>
        );
    }
}

export default App;
