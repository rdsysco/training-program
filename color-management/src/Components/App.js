import React, { Component } from 'react';
import AddColor from "./Color/AddColor";
import ColorList from "./Color/ColorList";
import Nanoid from 'nanoid';
import faker from 'faker';


class App extends Component {

  constructor() {
    super();

    this.state = {
      colors: []
    }

    for (let i = 0; i < 4; i++) {

      this.state.colors.push({
        id: Nanoid(),
        name: faker.commerce.color(),
        value: faker.internet.color(),
        rates: [Math.floor(Math.random() * 4) + 1],
      });
    
    }

    this.addColor = this.addColor.bind(this);
    this.deleteColor = this.deleteColor.bind(this);
    this.rateColor = this.rateColor.bind(this);
    this.editColor = this.editColor.bind(this);
  }

  updateState() {
    this.setState({
      colors: this.colors || []
    });
  }

  addColor(name, color) {

    let colors = Object.assign([], this.state.colors);
    
    colors.push({
      id: Nanoid(),
      rates: [],
      name: name,
      value: color
    });

    this.setState({
      colors: colors
    });
  }

  deleteColor(id) {

    let colors = Object.assign([], this.state.colors);

    colors = colors.filter((item) => item.id !== id);

    this.setState({
      colors: colors
    });
  }

  rateColor(id, value) {

    let colors = Object.assign([], this.state.colors);

    let color = colors.filter((item) => item.id === id)[0];

    if (color) {

      color.rates.push(value);

      this.setState({
        colors: colors
      });

    }
  }

  editColor(id, name, colorValue) {

    let colors = Object.assign([], this.state.colors);

    let color = colors.filter((item) => item.id === id);

    if (color) {

      color.name = name;
      color.value = colorValue;

      this.setState({
        colors: colors
      });

    }

  }

  render() {
    let colors = (this.state || {}).colors || [];

    return (
      <div className="App">

        <AddColor addHandler={this.addColor} />

        <ColorList colors={colors} deleteHandler={this.deleteColor} rateHandler={this.rateColor} />

      </div>
    );
  }
}

export default App;
