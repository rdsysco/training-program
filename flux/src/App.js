import React, { Component } from 'react';
import { connect } from 'react-redux';
import Header from './components/Header';
import Grid from './components/Grid';
import Form from './components/Form';
import { getInitialNotes, addNewNote, removeNote } from './store/actions';

class App extends Component {
  render() {
    return (
      <div>
        <Header name={this.props.name}/>
        <Form addNewNote={this.props.addNewNote1}/>
        <Grid notes={this.props.notes} removeNote={this.props.removeNote1}/>
      </div>
    );
  }
}

const mapDispatchProps = (dispatch, ownProps) => {
  return {
    getInitialNotes1: () => {
      dispatch(getInitialNotes())
    },
    addNewNote1: (note) => {
      dispatch(addNewNote(note))
    },
    removeNote1: (id) => {
      dispatch(removeNote(id))
    }
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    notes: state.notes,
    name: state.name
  }
}



export default connect(mapStateToProps, mapDispatchProps)(App);
