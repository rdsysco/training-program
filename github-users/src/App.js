import React, { Component } from 'react';
import { Route, BrowserRouter as Router, Switch, Link } from 'react-router-dom';
import UserIcon from './assets/images/masood.jpg';
import {Details} from './components/Details';
import {Home} from './components/Home';
import {NotFound} from './components/NotFound';
import {User} from './components/User';

class App extends Component {
  render() {
    return (
      <Router>
        <header>
          <a className="user-logo" 
             href="https://github.com/abdolian">
            <img src={UserIcon}/>
          </a>
          <label className="title">
            Masood Abdolian
          </label>
          <Link to="/" 
                className="home-link">
            Go To Home
          </Link>
        </header>
        <Switch>
          <Route path="/"                           component={Home}     exact/>
          <Route path="/user/:username"             component={User}     exact/>
          <Route path="/user/:username/details/:id" component={Details}  exact/>
          <Route                                    component={NotFound}      />
        </Switch>
        <footer>
          Â©2019, Masood, Inc. All Rights Reserved.
        </footer>
      </Router>
    );
  }
}

export default App;
