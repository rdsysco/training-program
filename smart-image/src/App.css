.App{
  flex: 1 1 auto;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  max-width: 100%;
  position: relative;
}

.v-responsive {
  position: relative;
  overflow: hidden;
  flex: 1 0 auto;
  display: flex;
}

.v-responsive__sizer {
  transition: padding-bottom 0.2s cubic-bezier(0.25, 0.8, 0.5, 1);
  flex: 0 0 0px;
}

.v-image__image--cover {
  background-size: cover;
}

.v-image__image {
  background-repeat: no-repeat;
}

.v-image__image, .v-image__placeholder {
  z-index: -1;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}

.v-responsive__content {
  flex: 1 0 0px;
}