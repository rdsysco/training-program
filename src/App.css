.App {
  text-align: center;
  
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100px;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: left;
  font-size: calc(10px + 2vmin);
  color: white;
  padding: 10px
}
.App-content {
  margin: 5px
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
