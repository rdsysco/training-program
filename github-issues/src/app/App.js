import React from 'react';
import './App.css';
import IssueContainer from "./issues/IssuesContainer"

function App() {
    return (
        <div className="container w-75 border text-truncate"><IssueContainer /></div>
    )
}

export default App;