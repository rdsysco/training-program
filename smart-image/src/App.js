import React, { Component } from 'react';
import './App.css';
import { SmartImage } from './components/SmartImage';
import img from './linear.gif';

class App extends Component {
  constructor() {
    super();
    this.state = {
      num: 44, src: 'https://avatars3.githubusercontent.com/u/43126626?s=400&v=4'
    };
  }
  componentDidMount() {
    setTimeout(() => {
      //this.setState({ src: 'https://avatars3.githubusercontent.com/u/43126626?s=400&v=4' })
      //this.setState({ src: 'https://avatars2.githubusercontent.com/u/12845?s=96&v=4' })
    }, 5000);
  }
  render() {
    return (
      <div className="App">
              <SmartImage />
          {/* <br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a */}

        <SmartImage grid={1}
        delay={500}
          src="dddddd"
          aspectRatio={2.3}
          width="128px"
          height={128}
          intersectionDelay={500}
          src={this.state.src}
          lazySrc={img} />
                    {/* <br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a */}
                    
                    
        <SmartImage grid={1}
        contain={true}
        delay={500}
          src="dddddd"
          aspectRatio={2.3}
          width="128px"
          height={128}
          intersectionDelay={500}
          src={this.state.src}
          lazySrc={img} />

                    {/* <br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a<br/>a */}

      </div>
    );
  }
}

export default App;
