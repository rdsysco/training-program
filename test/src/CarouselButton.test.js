import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CarouselButton from './CarouselButton';
import App from './App';

configure({ adapter: new Adapter() });

const app = shallow(<App/>);

it('app is ok', () => {
    expect(app).toMatchSnapshot();
});

// describe('CarouselButton', () => {
//     it('renders a <button>', () => {
//         const wrapper = shallow(<CarouselButton />);
//         expect(wrapper.type()).toBe('button');
//     });
//     it('passes `children` through to the <button>', () => {
//         const text = 'Button text';
//         const wrapper = shallow(<CarouselButton>{text}</CarouselButton>);
//         expect(wrapper.prop('children')).toBe(text);
//     });
// });