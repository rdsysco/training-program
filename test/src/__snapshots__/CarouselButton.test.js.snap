// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app is ok 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn-add"
          onClick={[Function]}
        >
          add
        </button>,
        <div
          className="gift-list"
        />,
        <header
          className="App-header"
        >
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <p>
            Edit 
            <code>
              src/App.js
            </code>
             and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React
          </a>
        </header>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "add",
          "className": "btn-add",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "add",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "gift-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <p>
              Edit 
              <code>
                src/App.js
              </code>
               and save to reload.
            </p>,
            <a
              className="App-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn React
            </a>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Edit ",
                <code>
                  src/App.js
                </code>,
                " and save to reload.",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Edit ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "src/App.js",
                },
                "ref": null,
                "rendered": "src/App.js",
                "type": "code",
              },
              " and save to reload.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Learn React",
              "className": "App-link",
              "href": "https://reactjs.org",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Learn React",
            "type": "a",
          },
        ],
        "type": "header",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn-add"
            onClick={[Function]}
          >
            add
          </button>,
          <div
            className="gift-list"
          />,
          <header
            className="App-header"
          >
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <p>
              Edit 
              <code>
                src/App.js
              </code>
               and save to reload.
            </p>
            <a
              className="App-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn React
            </a>
          </header>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "add",
            "className": "btn-add",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "add",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "gift-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />,
              <p>
                Edit 
                <code>
                  src/App.js
                </code>
                 and save to reload.
              </p>,
              <a
                className="App-link"
                href="https://reactjs.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                Learn React
              </a>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Edit ",
                  <code>
                    src/App.js
                  </code>,
                  " and save to reload.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Edit ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "src/App.js",
                  },
                  "ref": null,
                  "rendered": "src/App.js",
                  "type": "code",
                },
                " and save to reload.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Learn React",
                "className": "App-link",
                "href": "https://reactjs.org",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Learn React",
              "type": "a",
            },
          ],
          "type": "header",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
